plugins/PrivacyManager/Tracker/RequestProcessor.php

```diff
@@ -37,15 +38,25 @@ public function manipulateRequest(Request $request)
         }
     }
 
+    /**
+     * pseudo anonymization as we need to make sure to always generate the same UserId for the same original UserID
+     *
+     * @param $userId
+     * @return string
+     */
     public static function anonymizeUserId($userId)
     {
-        // pseudo anonymization as we need to make sure to always generate the same UserId for the same original UserID
-        return sha1($userId . SettingsPiwik::getSalt());
+        $trackerCache = Tracker\Cache::getCacheGeneral();
+        if (!empty($trackerCache[PrivacyManager::OPTION_USERID_SALT])) {
+            $salt = $trackerCache[PrivacyManager::OPTION_USERID_SALT];
+        } else {
+            $salt = PrivacyManager::getUserIdSalt();
```

Comment:
Ideally the else was avoided and in case it is not present, simply the raw `userId` removed and we could install maybe the salt with an update, and otherwise it gets installed through the tracker cache creation.